workflow/common/util.go

```diff
@@ -37,7 +37,8 @@ func FindOverlappingVolume(tmpl *wfv1.Template, path string) *apiv1.VolumeMount
 	var volMnt *apiv1.VolumeMount
 	deepestLen := 0
 	for _, mnt := range tmpl.GetVolumeMounts() {
-		if path != mnt.MountPath && !strings.HasPrefix(path, mnt.MountPath+"/") {
+		mnt := mnt
```

Comment:
I've discovered this additional bug by adding an extra volume and run the test. If not scoped correctly, `mnt` would change value of `volMnt`.

Reply:
I don't understand this line - it appears to be a no-op?

Reply:
It appears, however `mnt` *will* change value in the for loop. As it is copied by reference, the actual value will be changed once `volMnt` is returned.

This can be tested by leaving the old code but adding multiple volumeMounts in the test.

Reply:
With the line `mnt := mnt` we are changing scope of the `mnt` variable and thus creating an new variable with different reference (that will be copied later)

Reply:
Could we at least have the second `mnt` have a different name?
```suggestion
		mount := mnt
```

Or something?

Reply:
this seems odd to me, not something I would expect Go to do, not saying it is wrong, just that it unconventional. We want the deepest mount yes, so how about we sort the mounts by `len(mnt.MountPath)` (longest first) and then use the first that matches